specs/ServerCertificate.md

```diff
@@ -0,0 +1,264 @@
+    }
+    _isServerCertificateError = !_isServerCertificateError;
+
+    MessageBox.Show(this,
+      _isServerCertificateError ? "Custom server certificate error has been enabled" : "Custom server certificate error has been disabled",
+      "Custom server certificate error");
+  }
+  catch (NotImplementedException exception)
+  {
+    MessageBox.Show(this, "Custom server certificate error Failed: " + exception.Message, "Custom server certificate error");
+  }
+}
+
+void WebView_ReceivingServerCertificateError(object sender, CoreWebView2ReceivingServerCertificateErrorEventArgs e)
+{
+  CoreWebView2Certificate certificate = e.ServerCertificate;
+
+  // Continues with the navigation to a server with a TLS certificate if
+  // the certificate is invalid and signed by badssl authority.
+  if (e.ErrorStatus == CoreWebView2WebErrorStatus.CertificateIsInvalid &&
+      (certificate.Issuer == "*.badssl.com" ||
+      certificate.Issuer == "BadSSL Untrusted Root Certificate Authority"))
+  {
+    // Continue the request with the TLS certificate.
+    e.Handled = true;
+  }
+  else
+  {
+    // Cancel the request.
+    e.Cancel = true;
+  }
+}
+
+// This example clears the TLS decision in response to proceeding with TLS certificate errors.
+void ClearServerCertificateErrorOverrideCache()
+{
+  await webView.CoreWebView2.ClearServerCertificateErrorOverrideCacheAsync();
+  MessageBox.Show(this, "Cleared", "Clear server certificate error override cache");
+}
+```
+
+# API Details
+
+## Win32 C++
+``` cpp
+[uuid(4B7FF0D2-8203-48B0-ACBF-ED9CFF82567A), object, pointer_default(unique)]
+interface ICoreWebView2_11 : ICoreWebView2_10 {
+  /// Add an event handler for the ServerCertificateError event.
+  /// The ServerCertificateError event is raised when the WebView2
+  /// cannot verify server's digital certificate while loading a web page.
+  ///
+  /// With this event you have several options for responding to server
+  /// certificate error requests:
+  ///
+  /// Scenario                                                   | Handled | Cancel
+  /// ---------------------------------------------------------- | ------- | ------
+  /// Ignore the warning and continue the request                | True    | False
+  /// Cancel the request                                         | n/a     | True
+  /// Display default TLS interstitial page                      | False   | False
+  ///
+  /// If you don't handle the event, WebView2 will show the default TLS interstitial page to user.
+  ///
+  /// WebView2 caches the response when `Handled` is set `TRUE` for the host and certificate in the session
+  /// and ServerCertificateError event won't be raised again.
+  ///
+  /// To raise the event again you must clear the cache using `ClearServerCertificateErrorOverrideCache`.
+  ///
+  /// \snippet SettingsComponent.cpp ServerCertificateError1
+  HRESULT add_ReceivingServerCertificateError(
```

Comment:
It would be nice to say what events come just before or after this one. Like for a navigation, when does this event get raised relative to the other [navigation events](https://docs.microsoft.com/en-us/microsoft-edge/webview2/concepts/navigation-events)? Or for WebResourceRequested, I guess ReceivingServerCertificateError  comes just right before the WebResourceRequested event is raised?

Reply:
Added, this event follows `NavigationStarting` and if it continues, all the other navigation events are called else calls `NavigationCompleted` with failure message.